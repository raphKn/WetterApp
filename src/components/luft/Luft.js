import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, View } from 'react-native';
import CustomSection from '../common/CustomSection';
import Luftfeuchtigkeit from './Luftfeuchtigkeit';
import Luftdruck from './Luftdruck';
import Bewoelkung from './Bewoelkung';
import { isEmptyObject } from '../../utils/ObjectUtils';

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',

    },
});

const Luft = (props) => {
    if (isEmptyObject(props.luft)) {
        return null;
    }
    return (
        <CustomSection style={styles.container} name="Wolken und Luft">
            <Bewoelkung bewoelkung={props.luft.bewoelkung} />
            <View>
                <Luftfeuchtigkeit luftfeuchtigkeit={props.luft.luftfeuchtigkeit} />
                <Luftdruck luftdruck={props.luft.luftdruck} />
            </View>
        </CustomSection>
    );
};

Luft.propTypes = {
    luft: PropTypes.shape({
        bewoelkung: PropTypes.number,
        luftdruck: PropTypes.number,
        luftfeuchtigkeit: PropTypes.number,
    }),
};

Luft.defaultProps = {
    luft: null,
};

export default Luft;
