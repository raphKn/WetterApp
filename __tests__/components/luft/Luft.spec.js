import React from 'react';
import renderer from 'react-test-renderer';
import Luft from '../../../src/components/luft/Luft';

describe('Luft', () => {
    it('renders low bewoelkung correctly in a snapshot', () => {
        const luft = { bewoelkung: 30, luftdruck: 1.012, luftfeuchtigkeit: 40 };
        const wrapper = renderer.create(
            <Luft luft={luft} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders high bewpelkung correctly in a snapshot', () => {
        const luft = { bewoelkung: 70, luftdruck: 1.012, luftfeuchtigkeit: 40 };
        const wrapper = renderer.create(
            <Luft luft={luft} />);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders nothing for no luft in a snapshot', () => {
        const wrapper = renderer.create(
            <Luft luft={{}} />);
        expect(wrapper).toMatchSnapshot();
    });
});
