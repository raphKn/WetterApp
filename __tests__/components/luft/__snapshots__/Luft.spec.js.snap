// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Luft renders high bewpelkung correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "#3770a5",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#FAFAFA",
          },
          Object {
            "fontSize": 18,
            "paddingHorizontal": 18,
          },
        ]
      }
    >
      Wolken und Luft
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": null,
            "justifyContent": "center",
            "width": null,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": null,
                "width": null,
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          70
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        % bewölkt
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          40
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          % Luftfeuchtigkeit
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          1.012
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          hPa Luftdruck
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Luft renders low bewoelkung correctly in a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "paddingTop": 12,
      },
      Object {
        "backgroundColor": "#3770a5",
      },
    ]
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#FAFAFA",
          },
          Object {
            "fontSize": 18,
            "paddingHorizontal": 18,
          },
        ]
      }
    >
      Wolken und Luft
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E0E0E0",
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "marginHorizontal": 6,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": null,
            "justifyContent": "center",
            "width": null,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": null,
                "width": null,
              },
              undefined,
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          30
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#212121",
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        % bewölkt
      </Text>
    </View>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          40
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          % Luftfeuchtigkeit
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "fontSize": 32,
              },
            ]
          }
        >
          1.012
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          disabled={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#212121",
              },
              Object {
                "alignSelf": "center",
              },
            ]
          }
        >
          hPa Luftdruck
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Luft renders nothing for no luft in a snapshot 1`] = `null`;
